lib/analytics.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs');
+
+// We can either use a process-wide object regularly saved to a JSON file,
+// or a Redis equivalent (for multi-process / when the filesystem is unreliable.
+var redis;
+var useRedis = false;
+if (process.env.REDISTOGO_URL) {
+  var redisToGo = require('url').parse(process.env.REDISTOGO_URL);
+  redis = require('redis').createClient(redisToGo.port, redisToGo.hostname);
+  redis.auth(redisToGo.auth.split(':')[1]);
+  useRedis = true;
+}
+
+var analytics = {};
+
+var analyticsAutoSaveFileName = process.env.SHIELDS_ANALYTICS_FILE || './analytics.json';
+var analyticsAutoSavePeriod = 10000;
+setInterval(function analyticsAutoSave() {
+  if (useRedis) {
+    redis.set(analyticsAutoSaveFileName, JSON.stringify(analytics));
+  } else {
+    fs.writeFileSync(analyticsAutoSaveFileName, JSON.stringify(analytics));
```

Comment:
I'd suggest using the async version (`writeFile`) so that slow disk I/O doesn't block the thread.